### ISSUE TITLE
[BUG] moving average calculation of tot_loss metrics is wrong.

### ISSUE THREAD

**Issue by chenjiasheng** [k2-fsa/icefall#1009]
https://github.com/k2-fsa/icefall/blob/05e7435d0d36789ba947b3ca664b730b8d79cb92/icefall/transformer_lm/train.py#L416

the formular of moving average calculation should be:
`y = (1-k) * y + k * x` with `1-k` being the keep propotion.

so the mentioned line should be changed to:
`tot_loss = (tot_loss * (1 - 1 / params.reset_interval)) + loss_info / params.reset_interval`

This issue would cause loss/metric curve to vibrate abnormally.
Other train.py scripts such as those in dir `egs/*` also have this issue.
